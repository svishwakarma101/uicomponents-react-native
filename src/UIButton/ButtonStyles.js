import { StyleSheet } from 'react-native'
import { APPLIED_THEME as Theme } from '../utils/Constants'
import { Fonts, FontSize, Colors } from '../utils/StyleSheet'
import {getAdjustedFontSize} from '../utils/Fonts'

export const ButtonStyles = StyleSheet.create({
    primaryButton: {
      height: 46,
      backgroundColor: Theme.PrimaryButton.backgroundColor,
      borderRadius: 4,
      alignSelf: 'stretch',
      justifyContent: 'center',
      borderColor: Theme.PrimaryButton.borderColor,
    },
    secondaryButton: {
      height: 46,
      backgroundColor: Theme.SecondaryButton.backgroundColor,
      borderWidth: 2,
      borderColor: Theme.SecondaryButton.borderColor,
      borderRadius: 4,
      alignSelf: 'stretch',
      justifyContent: 'center',
    },
    disabledButton: {
      height: 46,
      backgroundColor: Theme.DisabledPrimaryButton.backgroundColor,
      borderWidth: 1,
      borderColor: Theme.DisabledPrimaryButton.borderColor,
      borderRadius: 4,
      alignSelf: 'stretch',
      justifyContent: 'center',
    },
    disabledSecondaryButton: {
      height: 46,
      backgroundColor: Theme.DisabledSecondaryButton.backgroundColor,
      borderColor: Theme.DisabledSecondaryButton.borderColor,
      borderRadius: 4,
      alignSelf: 'stretch',
      justifyContent: 'center',
      borderWidth: 2,
      fontWeight: 'bold',
    },
    gradientButton: {
      height: 60,
      backgroundColor: Theme.GradientButton.backgroundColorGradientStart,
      borderRadius: 4,
      alignSelf: 'stretch',
      justifyContent: 'center',
    },
    transparentButton: {
      flex: 0,
      backgroundColor: Theme.TransparentButton.backgroundColor,
      borderWidth: 0,
      borderColor: Theme.TransparentButton.borderColor,
      borderRadius: 0,
      alignSelf: 'stretch',
      justifyContent: 'center'
    },
    defaultButtonType: {
      height: 46,
      backgroundColor: Theme.DefaultButton.backgroundColor,
      borderWidth: 1,
      borderColor: Theme.DefaultButton.borderColor,
      alignSelf: 'stretch',
      justifyContent: 'center'
    },
    linkButtonStyle: {
      height: 44,
      alignItems: 'center',
      alignSelf: 'center',
      paddingHorizontal: 20,
      backgroundColor: Colors.transparent,
      borderColor: Colors.transparent,
      borderWidth: 0,
    },
    defaultShapeButton: {
      borderRadius: 4,
    },
    roundedEdgeButton: {
      borderRadius: 10,
    },
    roundButton: {
      borderRadius: 30,
    },
    squareButton: {
      borderRadius: 0,
    },
    linkButtonTextStyle: {
      // fontFamily: Fonts.BrandSemibold,
      fontSize: getAdjustedFontSize(FontSize.Medium),
      color: Colors.linkTextColor
    },
    title: {
      textAlign: 'center',
      // fontFamily: Fonts.Heavy,
      fontSize: FontSize.Medium,
      backgroundColor: 'transparent',
      fontWeight: 'bold'
    },
    titleView: {
      justifyContent: 'center',
      marginHorizontal: '10%'
    },
    contentWithIcon: {
      alignItems: 'center',
      justifyContent: 'center',
      marginLeft: '10%',
      marginRight: '10%'
    }
  })