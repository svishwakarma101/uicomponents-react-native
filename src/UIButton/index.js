import React, { useState } from 'react'
import { View, Text, TouchableOpacity, ViewPropTypes, Image } from 'react-native'
import { BUTTON_TYPES, BUTTON_SHAPES } from '../utils/Constants'
import { APPLIED_THEME as Theme } from '../utils/Constants'
import renderIf from '../utils/RenderIf'
import { ButtonStyles } from './ButtonStyles'
import PropTypes from 'prop-types'
import LinearGradient from 'react-native-linear-gradient'
import { replaceSpaceWithUnderscore, accessibilityId } from '../utils/index'

function UIButton(props) {
  var { content,
    buttonType,
    style,
    buttonShape,
    disabled,
    gradientStyle,
    iconLeftPositioned,
    onPressIn,
    iconWithBtnText,
    titleStyle,
    iconStyle,
    contentStyle,
    onPressOut,
  linkStyle } = props
  const [buttonPressedIn, setButtonPressedIn] = useState(false)
  var buttonStyle
  var buttonShape = buttonShape || 'default'
  var buttonType = buttonType || 'default'
  var isDisabled = disabled || false
  var isIconLeftPositioned = iconLeftPositioned || false
  var buttonBackgroundColor
  var buttonBorderColor

  const isGradient = buttonType === BUTTON_TYPES.gradient;

  switch (buttonType) {
    case BUTTON_TYPES.primary:
      buttonStyle = ButtonStyles.primaryButton;
      break;
    case BUTTON_TYPES.secondary:
      buttonStyle = ButtonStyles.secondaryButton;
      break;
    case BUTTON_TYPES.disabled:
      buttonStyle = ButtonStyles.disabledButton;
      break;
    case BUTTON_TYPES.gradient:
      buttonStyle = ButtonStyles.gradientButton;
      break;
    case BUTTON_TYPES.transparent:
      buttonStyle = ButtonStyles.transparentButton;
      break;
    default:
      buttonStyle = ButtonStyles.defaultButtonType;
      break;
  }

  switch (buttonShape) {
    case BUTTON_SHAPES.square:
      buttonShape = ButtonStyles.squareButton;
      break;
    case BUTTON_SHAPES.rounded:
      buttonShape = ButtonStyles.roundButton;
      break;
    case BUTTON_SHAPES.roundedEdge:
      buttonShape = ButtonStyles.roundedEdgeButton;
      break;
    default:
      buttonShape = ButtonStyles.defaultShapeButton;
      break;
  }

  var textColor = getTextColor(buttonPressedIn, buttonType, isDisabled);
  var iconColor = getImageColor(buttonType, isDisabled)
  let buttonAttributes = { buttonStyle, buttonShape, textColor };

  if (!isGradient) {
    buttonBackgroundColor = getButtonBackgroundColor(buttonType, isDisabled);
    buttonBorderColor = getButtonBorderColor(buttonType, isDisabled);
    buttonAttributes = {
      ...buttonAttributes,
      buttonBackgroundColor,
      buttonBorderColor,
    };
  }

  if (buttonType === BUTTON_TYPES.link) {
    buttonAttributes = {
      buttonStyle: ButtonStyles.linkButtonStyle,
      buttonShape: null,
      textColor: ButtonStyles.linkButtonTextStyle.color,
      buttonBackgroundColor: 'transparent',
      buttonBorderColor: 'transparent',
      linkTextStyle: ButtonStyles.linkButtonTextStyle,
      linkStyle: linkStyle
    }
  }

  if (props.buttonType === BUTTON_TYPES.primary) {
    buttonAttributes = {
      ...buttonAttributes,
      titleTextStyle: props.isTitleCapital ? { textTransform: 'uppercase' } : { textTransform: 'capitalize' }
    }
  }

  return isGradient
    ? getGradientButton(buttonAttributes)
    : getNormalButton(buttonAttributes);

  function getGradientButton(buttonAttributes) {
    const { buttonStyle, buttonShape, textColor } = buttonAttributes;
    return (
      <View>
        <TouchableOpacity
          {...props}
          disabled={disabled || false}
          style={[buttonStyle, buttonShape, style]}
          activeOpacity={1.0}
          delayPressIn={100}
          delayPressOut={100}
          onPressIn={() => btnPressedIn()}
          onPressOut={() => btnPressedOut()}
          accessibilityLabel={accessibilityId(
            getButtonAccessibilityLabel(content || 'icon'), content
          )}
          testID={getButtonAccessibilityLabel(content || 'icon')}
        >
          <LinearGradient
            start={{ x: 0, y: 0 }}
            end={{ x: 1, y: 0 }}
            colors={[
              Theme.GradientButton.backgroundColorGradientStart,
              Theme.GradientButton.backgroundColorGradientMid,
              Theme.GradientButton.backgroundColorGradientEnd,
            ]}
            style={[
              buttonStyle,
              buttonShape,
              gradientStyle,
            ]}
          >
            {
              renderIf(content || iconWithBtnText)(
                <View style={[ButtonStyles.contentWithIcon, contentStyle && contentStyle,
                { flexDirection: isIconLeftPositioned ? 'row-reverse' : 'row' }]}>
                  {renderIf(content)(
                  <View style={[ButtonStyles.titleView]}>
                    <Text
                      style={[ButtonStyles.title,
                      { color: textColor },
                        titleStyle
                      ]}
                      numberOfLines={1}
                    > {content}
                    </Text>
                  </View>
                  )}
                  <View style={{ justifyContent: 'center' }}>
                    <Image source={iconWithBtnText} style={[{ tintColor: iconColor }, iconStyle]} />
                  </View>
                </View>
              )
            }
            {props.children}
          </LinearGradient>
        </TouchableOpacity>
      </View>
    )
  }

  function getNormalButton(buttonAttributes) {
    const {
      buttonStyle,
      buttonShape,
      textColor,
      buttonBackgroundColor,
      buttonBorderColor,
      linkTextStyle,
      titleTextStyle,
      linkStyle
    } = buttonAttributes;
    return (
      <View>
        <TouchableOpacity
          {...props}
          style={[
            buttonStyle,
            buttonShape,
            passStyles(buttonBackgroundColor, buttonBorderColor),
            style
          ]}
          disabled={disabled ? disabled : false}
          activeOpacity={1.0}
          delayPressIn={100}
          delayPressOut={100}
          onPressIn={() => btnPressedIn()}
          onPressOut={() => btnPressedOut()}
          accessibilityLabel={accessibilityId(
            getButtonAccessibilityLabel(content || 'icon'), content
          )}
          testID={getButtonAccessibilityLabel(content || 'icon')}
        >
          {renderIf(content || iconWithBtnText)(
            <View style={[ButtonStyles.buttonContainer, 
              iconWithBtnText && iconWithBtnText ? ButtonStyles.contentWithIcon : null, 
              contentStyle && contentStyle,
            { flexDirection: isIconLeftPositioned ? 'row-reverse' : 'row' }]}>
              {renderIf(content)(
              <View style={[buttonType === 'link' ? null : ButtonStyles.titleView]}>
                <Text
                  style={[
                    ButtonStyles.title,
                    { color: textColor },
                    linkTextStyle,
                    titleTextStyle,
                    titleStyle,
                    linkStyle
                  ]}
                  numberOfLines={1}
                >
                  {content}
                </Text>
              </View>
              )}
              <View style={{ justifyContent: 'center' }}>
                <Image source={iconWithBtnText} style={[{ tintColor: iconColor }, iconStyle]} />
              </View>
            </View>
          )}
          {props.children}
        </TouchableOpacity>
      </View>
    );
  }

  function passStyles() {
    return {
      backgroundColor: buttonBackgroundColor,
      borderColor: buttonBorderColor,
      opacity: buttonType === BUTTON_TYPES.disabled ? 0.5 : 1,
    }
  }

  function btnPressedIn() {
    if (!disabled) {
      setButtonPressedIn(true)
    }
    if (onPressIn !== undefined) {
      onPressIn()
    }
  }

  function btnPressedOut() {
    if (!disabled) {
      setButtonPressedIn(false)
    }
    if (onPressOut !== undefined) {
      onPressOut();
    }
  }
}

function getButtonBackgroundColor(buttonType, isDisabled) {
  const buttonTheme = getThemeForButtonType(buttonType, isDisabled)
  return buttonTheme.backgroundColor
}

function getTextColor(buttonPressedIn, buttonType, isDisabled) {
  const buttonTheme = getThemeForButtonType(buttonType, isDisabled)
  let textColor = buttonTheme.textColor;
  if (buttonPressedIn) {
    textColor = buttonTheme.activeTextColor;
  }
  return textColor
}

function getImageColor(buttonType, isDisabled) {
  const buttonTheme = getThemeForButtonType(buttonType, isDisabled)
  let iconColor = buttonTheme.tintColor
  return iconColor
}

function getThemeForButtonType(buttonType, isDisabled) {
  if (buttonType === BUTTON_TYPES.primary) {
    return isDisabled ? Theme.DisabledButton : Theme.PrimaryButton
  } else if (buttonType === BUTTON_TYPES.secondary) {
    return isDisabled ? Theme.DisabledSecondaryButton : Theme.SecondaryButton;
  } else if (buttonType === BUTTON_TYPES.disabled) {
    return Theme.DisabledButton;
  } else if (buttonType === BUTTON_TYPES.gradient) {
    return Theme.GradientButton;
  } else if (buttonType === BUTTON_TYPES.link) {
    return Theme.LinkButton
  }
  return Theme.DefaultButton;
}

function getButtonBorderColor(buttonType, isDisabled) {
  const buttonTheme = getThemeForButtonType(buttonType, isDisabled);
  return buttonTheme.borderColor;
}

function getButtonAccessibilityLabel(content) {
  let label = ''
  label = `${replaceSpaceWithUnderscore(content)}_button`
  return label
}
UIButton.propTypes = {
  theme: PropTypes.object,
  disabled: PropTypes.bool,
  buttonType: PropTypes.oneOf(Object.keys(BUTTON_TYPES)),
  buttonShape: PropTypes.oneOf(Object.keys(BUTTON_SHAPES)),
  titleStyle: PropTypes.shape({
    ...ViewPropTypes.style
  }),
  gradientStyle: PropTypes.shape({
    ...ViewPropTypes.style
  }),
  linkStyle: PropTypes.shape({
    ...ViewPropTypes.style
  }),
  contentStyle: PropTypes.shape({
    ...ViewPropTypes.style
  }),
  iconStyle: PropTypes.shape({
    ...ViewPropTypes.style
  }),
  iconLeftPositioned: PropTypes.bool,
  onPressIn: PropTypes.func,
  onPressOut: PropTypes.func
};

UIButton.defaultProps = {
  theme: Theme
};

export default UIButton