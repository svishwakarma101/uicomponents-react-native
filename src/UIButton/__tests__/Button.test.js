import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import UIButton from '..'
import sinon from 'sinon'

describe('UIButton ', () => {
  it('Renders UIButton for default scenarios correctly ', () => {
    let fieldValue = renderer.create(<UIButton/>)
    expect(fieldValue).toMatchSnapshot()
  });

  it('renders UIButton of type Primary correctly', () => {
    const component = renderer.create(<UIButton buttonType={'primary'}/>)
    expect(component).toMatchSnapshot()
  });

  it('renders UIButton of type Secondary correctly ', () => {
    const component = renderer.create(<UIButton buttonType={'secondary'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of type Disabled correctly ', () => {
    const component = renderer.create(<UIButton buttonType={'disabled'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of type Gradient correctly ', () => {
    const component = renderer.create(<UIButton buttonType={'gradient'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of type Transparent correctly ', () => {
    const component = renderer.create(<UIButton buttonType={'transparent'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of shape Square correctly ', () => {
    const component = renderer.create(<UIButton buttonShape={'square'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of shape Rounded correctly ', () => {
    const component = renderer.create(<UIButton buttonShape={'rounded'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of shape RoundedEdge correctly ', () => {
    const component = renderer.create(<UIButton buttonShape={'roundedEdge'} />);
    expect(component).toMatchSnapshot();
  });

  it('renders UIButton of prop disabled', () => {
    const component = renderer.create(<UIButton disabled/>)
    expect(component).toBeTruthy()
  });

  it('renders UIButton for prop leftIconPosition', () => {
    const component = shallow(<UIButton iconLeftPositioned/>)
    expect(component).toBeTruthy()
  })

  it('To verify UIButton actions are called when button is clicked', () => {
    const btnPressedIn = jest.fn()
    const btnPressedOut = jest.fn()

    const component = shallow(<UIButton onPressIn={btnPressedIn} onPressOut={btnPressedOut}/>)

    const button = component.find('TouchableOpacity');
    button.simulate('pressIn')
    expect(btnPressedIn).toHaveBeenCalled()

    button.simulate('pressOut');
    expect(btnPressedOut).toHaveBeenCalled()
  })

  it('To verify whether UIButton actions are called when disabled props is received', () => {
    const btnPressedIn=jest.fn();
    const btnPressedOut=jest.fn();

    const component= shallow(<UIButton disabled/>)
    
    const button = component.find('TouchableOpacity');
    button.simulate('pressIn')
    expect(btnPressedIn).not.toHaveBeenCalled()

    button.simulate('pressOut')
    expect(btnPressedOut).not.toHaveBeenCalled()
  })
})
