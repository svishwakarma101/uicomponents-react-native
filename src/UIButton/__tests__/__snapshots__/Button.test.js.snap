// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIButton  Renders UIButton for default scenarios correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FE0095",
        "borderColor": "#FE0095",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 46,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of shape Rounded correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FE0095",
        "borderColor": "#FE0095",
        "borderRadius": 30,
        "borderWidth": 1,
        "height": 46,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of shape RoundedEdge correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FE0095",
        "borderColor": "#FE0095",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 46,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of shape Square correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FE0095",
        "borderColor": "#FE0095",
        "borderRadius": 0,
        "borderWidth": 1,
        "height": 46,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of type Disabled correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#E2E2E2",
        "borderColor": "#E2E2E2",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 46,
        "justifyContent": "center",
        "opacity": 0.5,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of type Gradient correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#2974D2",
        "borderRadius": 4,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  >
    <BVLinearGradient
      colors={
        Array [
          4280906962,
          4280628372,
          4280233088,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#2974D2",
            "borderRadius": 4,
            "height": 60,
            "justifyContent": "center",
          },
          Object {
            "borderRadius": 4,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`UIButton  renders UIButton of type Primary correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FE0095",
        "borderColor": "#FE0095",
        "borderRadius": 4,
        "height": 46,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of type Secondary correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FE0095",
        "borderRadius": 4,
        "borderWidth": 2,
        "height": 46,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;

exports[`UIButton  renders UIButton of type Transparent correctly  1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FE0095",
        "borderColor": "#FE0095",
        "borderRadius": 4,
        "borderWidth": 0,
        "flex": 0,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon_button"
  />
</View>
`;
